### ISSUE TITLE
Citations in Rmarkdown

### ISSUE THREAD

**Issue by hansvancalster** [inbo/tutorials#207]
Given that RStudio will soon have new and improved functionalities to work with references/citations in Rmarkdown files, a tutorial on this topic should be worthwhile.

**Comment by florisvdh** [comment]
Current [preview](https://rstudio.com/products/rstudio/download/preview/) version of RStudio has the Visual Markdown Editor, so the functionality has become available. It appears that the preview is currently being updated on a very regular basis, presumably to iron out remaining problems.

It works fluently with a moderately large Zotero database (>3700 items). The citation feature is currently not available in source mode (rstudio/rstudio#7876).

Some notes about the recommended file format to store references:

- use of a bib(la)tex file (.bib) is supported by pandoc, but it is to be avoided rather than recommended (see also https://github.com/inbo/INBOmd/issues/62 and the link therein). It is less supported than following alternatives.
- the native, official format for use with [CSL](https://docs.citationstyles.org/en/1.0.1/specification.html) style files (and any [CSL processor](https://citationstyles.org/developers/#/csl-processors), especially the reference implementation _citeproc-js_) is CSL-JSON. [Here](https://www.google.com/url?q=https://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html) is its documentation.
- while CSL-JSON essentially _is_ human-readable json, pandoc has also implemented an alternative CSL-YAML standard, which is much more readable (to the same level as bibtex I believe). pandoc supports conversions between these formats (see the [citations section](https://pandoc.org/MANUAL.html#citations) in the manual of pandoc's markdown). Hence I believe CSL-YAML is restricted to use with pandoc only, although it can also be imported in e.g. Zotero by using the [BBT](https://retorque.re/zotero-better-bibtex/) addon.
- in short, only CSL-JSON and CSL-YAML guarantee lossless rendering of citations & bibliography in R markdown (which uses pandoc). And RStudio allows adding selected references to such files through a GUI.
- multiple reference files and formats can be combined in the same Rmd file using this format for the `bibliography` key in the YAML header:

```
bibliography: [references.json, references.yaml]
```

**Comment by hansvancalster** [comment]
Thanks for these insights and the news @florisvdh. 
I think we better wait a while with this tutorial until RStudio 1.4 is officially released and effectively installed @ INBO.

**Comment by florisvdh** [comment]
Sure, it was planned for 2021. I just wanted to drop some notes.

**Comment by florisvdh** [comment]
Update: RStudio 1.4 (with VME) has been released in January. See also https://blog.rstudio.com/2021/01/19/announcing-rstudio-1-4/
